lib/jekyll-admin/static_directory.rb

```diff
@@ -0,0 +1,89 @@
+module JekyllAdmin
+  class StaticDirectory
+
+    include Enumerable
+    include JekyllAdmin::URLable
+    include JekyllAdmin::APIable
+
+    DIRECTORY_TYPE = 'directory'
+    DIRECTORY_SEPARATOR = '/'
+
+    def initialize(path, config={})
+      @base = Pathname.new config.fetch(:base_directory)
+      @path           = Pathname.new path
+      @entries        = initialize_entries
```

Comment:
Rather than reading the directory on init, I'd have a dedicated `entries` method that's memorialized when first called.